rx-preferences/src/main/java/com/f2prateek/rx/preferences/RxSharedPreferences.java

```diff
@@ -93,10 +93,11 @@ public void onSharedPreferenceChanged(SharedPreferences preferences, String key)
     return new Preference<>(preferences, key, defaultValue, FloatAdapter.INSTANCE, keyChanges);
   }
 
-  /** Create an integer preference for {@code key}. Default is {@code null}. */
+  /** Create an integer preference for {@code key}. Default is {@code 0}. */
   @CheckResult @NonNull
   public Preference<Integer> getInteger(@NonNull String key) {
-    return getInteger(key, null);
+    //noinspection UnnecessaryBoxing
+    return getInteger(key, Integer.valueOf(0));
```

Comment:
Yup, should probably use `Boolean.valueOf` above. But probably want to cache Float ourselves, since it always returns a [new instance](https://android.googlesource.com/platform/libcore/+/refs/heads/master/luni/src/main/java/java/lang/Float.java).